src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioWorkspaceStatusServiceFactory.cs

```diff
@@ -77,17 +78,23 @@ public Service(IAsyncServiceProvider2 serviceProvider, IAsynchronousOperationLis
 
             public async Task WaitUntilFullyLoadedAsync(CancellationToken cancellationToken)
             {
-                await EnsureInitializationAsync(cancellationToken).ConfigureAwait(false);
-
-                var status = await GetProgressStageStatusAsync(cancellationToken).ConfigureAwait(false);
-                if (status == null)
+                using (Logger.LogBlock(FunctionId.PartialLoad_FullyLoaded, KeyValueLogMessage.NoProperty, cancellationToken))
                 {
-                    return;
-                }
+                    await EnsureInitializationAsync(cancellationToken).ConfigureAwait(false);
 
-                // TODO: WaitForCompletionAsync should accept cancellation directly.
-                //       for now, use WithCancellation to indirectly add cancellation
-                await status.WaitForCompletionAsync().WithCancellation(cancellationToken).ConfigureAwait(false);
+                    var status = await GetProgressStageStatusAsync(cancellationToken).ConfigureAwait(false);
+                    if (status == null)
+                    {
+                        return;
+                    }
+
+                    var completionTask = status.WaitForCompletionAsync();
+                    Logger.Log(FunctionId.PartialLoad_FullyLoaded, KeyValueLogMessage.Create(LogType.Trace, m => m["InitiallyFullyLoaded"] = completionTask.IsCompleted));
```

Comment:
Minor - would maybe rename to 'AlreadyFullyLoaded' ?

Reply:
sure.